docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1297 @@
+  routes on each transit router distributed half on the node.
+* In future there is a requirement to not create OVN constructs for UDNs
+  on those nodes where we know there won't be pod's scheduled. This is
+  to optimize resource consumption on large scale clusters with 1000 nodes
+  where only 10 nodes are going to host UDNs. Whatever topology we pick
+  here must work for that future enhancement as changing topologies is not
+  really encouraged.
+* The annotations we add to the CNC has cost of marshall and unmarshall
+  at each ovnkube-controller
+
+![blue-green-yellow-l3-connect-scale](images/l3-connecting-udns-scale.png)
+
+## OVN Kubernetes Version Skew
+
+Targeting release 1.2
+
+## Alternatives
+
+### Using BGP and RouterAdvertisement API
+
+If admin wants to connect two UDNs together then they can simply expose
+the two UDNs using RouterAdvertisement CR and the router acting as
+gateway outside the cluster will learn the pod subnet routes from both
+these UDNs thereby acting as the router and being able to connect pods
+together. Today we add ACLs, but those could be removed to accommodate
+the connectivity.
+
+**Pros**
+
+* No need for a new API to connect UDNs together
+* Implementation changes are also almost nil
+
+**Cons**
+
+* Some users might want to use BGP and UDNs only for the pods to be
+  directly accessible from external destinations and within the cluster
+  they might still want to respect segmentation - this is not possible
+  if we declare exposing UDNs subnets using BGP also means they are now
+  connected together. So basically the current API didn't account for
+  BGP to solve connecting UDNs problem.
+* Some users might not want to use BGP at all because that makes
+  assumptions on having the right fabric at the cluster infrastructure
+  layer like presence of FRR-K8s which might not be the case always -
+  then how can they connect their UDNs together? Solution needs to be
+  OVN-native where-ever possible.
+* Users cannot ask for partial v/s full connectivity across these UDNs
+  - example allow only services not pods.
+* Hardware offload not possible since traffic is not curtailed only to
+  the OVS stack
+
+### Using alternative connection ideas on OVN topology level
+
+See [discarded-alternatives.md](discarded-alternatives.md) for details
+on alternative topology connection ideas that were considered but
+discarded.
+
+### Using NodeNetwork CR instead of Annotations on Connect API
+
+NOTE: Having the `NodeNetwork` CRD as an internal data storage API might
+be more ideal to store the above information internally as this needs to
+be persisted into etcd on reboots. It will simplify the annotations and
+also help with scale since there are known issues around patching of
+annotation and waiting for reading and unmarshalling the annotation
+impacts scale. But this theory is not proven yet and there was a concern
+of nodes getting polluted with events?? (TBD: need to think why we didn't
```

Comment:
@npinaeva can you PTAL this and if you remember what the concerns were around scale for this option- I remember we clearly went with annotations because nodenetwork crd had a downside that I now can't remember.